docs/topics/native/native-app-with-c-and-libcurl.md

```diff
@@ -0,0 +1,260 @@
+        executable { 
+            entryPoint = "main"
+        }
+    }
+}
+    
+```      
+
+```groovy
+nativeTarget.with {
+    compilations.main { // NL
+        cinterops {     // NL
+            libcurl     // NL
+        }               // NL
+    }                   // NL
+    binaries {
+        executable {
+            entryPoint = 'main'
+        }
+    }
+}
+```
+
+</tabs>
+
+The new lines are marked with `// NL`. First, `cinterops` is added, and then an entry for each `def` file. By default,
+the name of the file is used. You can override this with additional parameters:
+
+```groovy
+libcurl {
+    defFile project.file("libcurl.def")
+    packageName 'com.jetbrains.handson.http'
+    compilerOpts '-I/path'
+    includeDirs.allHeaders("path")
+}
+```
+
+See the [Interoperability with C](native-c-interop.md) section for more details on the available options.
+
+## Write the application code
+
+Now you have the library and the corresponding Kotlin stubs and can consume them from your application. For this tutorial,
+let's convert the [simple.c](https://curl.haxx.se/libcurl/c/simple.html) example to Kotlin.
+
+In the `src/nativeMain/kotlin/` folder, update your `Main.kt` file with the following code:
+
+```kotlin
+import kotlinx.cinterop.*
+import libcurl.*
+
+fun main(args: Array<String>) {
+    val curl = curl_easy_init()
+    if (curl != null) {
+        curl_easy_setopt(curl, CURLOPT_URL, "https://example.com")
+        curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L)
+        val res = curl_easy_perform(curl)
+        if (res != CURLE_OK) {
+            println("curl_easy_perform() failed ${curl_easy_strerror(res)?.toKString()}")
+        }
+        curl_easy_cleanup(curl)
+    }
+}
+```
+
+As you can see, explicit variable declarations are eliminated in the Kotlin version, but everything else is pretty much
+the same as the C version. All the calls you'd expect in the `libcurl` library are available in the Kotlin equivalent.
+
+> This is a line-by-line literal translation. You could also write this in a more Kotlin idiomatic way.
+>
+
+## Compile and run the application
+
+1. Compile the application. To do that, run the following command in the terminal:
+
+```bash
+./gradlew runDebugExecutableNative
+```
+In this case, the `cinterop` generated part is implicitly included in the build.
+
+2. If there are no errors during compilation, run the build by creating a [Run Configuration](https://www.jetbrains.com/help/idea/run-debug-configuration.html#create-permanent).
+IntelliJ IDEA opens the **Run** tab and shows the output — the contents of `https://example.com`:
+
+![Application output with HTML-code](native-output.png){width="700"}
+
+You can see the actual output because the call `curl_easy_perform` prints the result to the standard output. You could
+hide this using `curl_easy_setopt`.
+
+> You can find the resulting code in our [GitHub](https://github.com/Kotlin/kotlin-hands-on-intro-kotlin-native) repository.
+>
```

Comment:
TBH, I don't understand the following:
here and there type = "note" line has been omitted.

What was the motivation for this?

Reply:
To make a different type of note, a tip. This way the article looks less monotonous